File: src/main/java/ru/markn/webchat/repository/UserRepository.java
package ru.markn.webchat.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.markn.webchat.models.User;

@Repository
public interface UserRepository extends JpaRepository<User,Long>
{
    User findByUserName(String userName);
}

File: src/main/kotlin/ru/markn/webchat/repositories/UserRepository.kt
package ru.markn.webchat.repositories

import org.springframework.data.jpa.repository.JpaRepository
import ru.markn.webchat.models.User
import java.util.Optional

interface UserRepository : JpaRepository<User, Long> {
    fun findByUsername(username: String): Optional<User>
    fun findUsersByUsernameIn(usernames: List<String>): List<User>
    fun findUsersByUsernameContains(username: String): List<User>
    fun findByEmail(email: String): Optional<User>
    fun existsByUsername(username: String): Boolean
    fun existsByEmail(email: String): Boolean
}